### PR TITLE
parametrics/apps/golang: call flush instead of stop in tracestats tests

### LINKED ISSUE #596: parametric/apps/golang: Flush() doesn't flush stats
The `FlushTraceStats` function for the Go implementation should be able to call `Flush()`, but this function doesn't actually flush the trace stats. This causes problems for the some of the tests which are then forced to call `Stop()`.

https://github.com/DataDog/system-tests/pull/576 will include some stopgaps, but we need to investigate if the Go Tracer should actually be flushing stats in `Flush()`

/cc @Kyle-Verhoog

### PR DESCRIPTION
**Pull request by lievan**
## Description

Previously, trace stats tests for golang had to call `stop` in order to flush trace stats as a stopgap for #596. After fixes from [DataDog/dd-trace-go#1661](https://github.com/DataDog/dd-trace-go/pull/1661), tests can now call `flush`. 

Fixes #596 

This PR also updates go.mod in parametric/apps/golang to point to [DataDog/dd-trace-go#1661](https://github.com/DataDog/dd-trace-go/pull/1661) which is now merged.
## Check list

Your PR is not ready to be reviewed? Please save it as a draft :pray:

- [ ] Follows the style guidelines of this project (See [how to easily lint the code](https://github.com/DataDog/system-tests/blob/main/docs/edit/lint.md))
- [ ] CI is passing
- [ ] There is at least one approval from code owners

Yes to all? Feel free to merge it whenever you want :heart:

### CODE REVIEW

File: parametric/test_library_tracestats.py [hunk 1]
```diff
@@ -151,8 +151,7 @@ def test_distinct_aggregationkeys_TS003(library_env, test_agent, test_library, t
             span.set_meta(key="http.status_code", val="400")
 
     if test_server.lang == "golang":
```
Comment (Kyle-Verhoog) [inline parametric/test_library_tracestats.py:153]: can we remove this check completely? `flush` should be called by the test fixture

**Review by Kyle-Verhoog — COMMENTED** [review]
great! just one question, but looks good to me

File: parametric/test_library_tracestats.py (hunk 1 above)
Comment (lievan) [reply parametric/test_library_tracestats.py:153]: Ah yes, just removed the checks!

**Review by Kyle-Verhoog — APPROVED** [review]
great! thank you 🙌

**Review by pawelchcki — APPROVED** [review]

**Pull request merged**